pkg/render/whisker/component.go

```diff
@@ -0,0 +1,427 @@
+				"k8s-app": WhiskerDeploymentName,
+			},
+		},
+	}
+}
+
+func (c *Component) guardianContainer() corev1.Container {
+	tunnelCAType := c.cfg.ManagementClusterConnection.Spec.TLS.CA
+	voltronURL := c.cfg.ManagementClusterConnection.Spec.ManagementClusterAddr
+	bundle := c.cfg.TrustedCertBundle
+
+	return corev1.Container{
+		Name:            GuardianContainerName,
+		Image:           c.guardianImage,
+		ImagePullPolicy: render.ImagePullPolicy(),
+		Env: append([]corev1.EnvVar{
+			{Name: "GUARDIAN_PORT", Value: "9443"},
+			{Name: "GUARDIAN_LOGLEVEL", Value: "DEBUG"},
+			{Name: "GUARDIAN_VOLTRON_URL", Value: voltronURL},
+			{Name: "GUARDIAN_VOLTRON_CA_TYPE", Value: string(tunnelCAType)},
+			{Name: "GUARDIAN_PACKET_CAPTURE_CA_BUNDLE_PATH", Value: bundle.MountPath()},
+			{Name: "GUARDIAN_PROMETHEUS_CA_BUNDLE_PATH", Value: bundle.MountPath()},
+			{Name: "GUARDIAN_QUERYSERVER_CA_BUNDLE_PATH", Value: bundle.MountPath()},
+		}, c.cfg.Installation.Proxy.EnvVars()...),
+		SecurityContext: securitycontext.NewNonRootContext(),
+		VolumeMounts: append([]corev1.VolumeMount{
+			secretMount("/certs", c.cfg.TunnelSecret),
+		}, c.cfg.TrustedCertBundle.VolumeMounts(rmeta.OSTypeLinux)...),
+	}
+}
+
+func (c *Component) deployment() *appsv1.Deployment {
+	tolerations := append(c.cfg.Installation.ControlPlaneTolerations, rmeta.TolerateCriticalAddonsAndControlPlane...)
+	if c.cfg.Installation.KubernetesProvider.IsGKE() {
+		tolerations = append(tolerations, rmeta.TolerateGKEARM64NoSchedule)
+	}
+
+	ctrs := []corev1.Container{c.whiskerContainer(), c.whiskerBackendContainer(), c.goldmaneContainer()}
+	volumes := []corev1.Volume{c.cfg.TrustedCertBundle.Volume()}
+	if c.cfg.ManagementClusterConnection != nil {
+		ctrs = append(ctrs, c.guardianContainer())
+		volumes = append(volumes, secretVolume(c.cfg.TunnelSecret))
+	}
+
+	if c.cfg.LinseedPublicCASecret != nil {
+		volumes = append(volumes, secretVolume(c.cfg.LinseedPublicCASecret))
+	}
+
+	return &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{Kind: "Deployment", APIVersion: "apps/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      WhiskerDeploymentName,
+			Namespace: WhiskerNamespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: ptr.ToPtr(int32(1)),
+			Strategy: appsv1.DeploymentStrategy{
+				Type: appsv1.RecreateDeploymentStrategyType,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: WhiskerDeploymentName,
+				},
+				Spec: corev1.PodSpec{
+					NodeSelector:       c.cfg.Installation.ControlPlaneNodeSelector,
+					ServiceAccountName: WhiskerServiceAccountName,
+					Tolerations:        tolerations,
+					ImagePullSecrets:   secret.GetReferenceList(c.cfg.PullSecrets),
+					Containers:         ctrs,
+					Volumes:            volumes,
+				},
+			},
+		},
+	}
+}
+
+func secretMount(path string, scrt *corev1.Secret) corev1.VolumeMount {
+	return corev1.VolumeMount{
+		Name:      fmt.Sprintf("%s-%s", scrt.Name, "scrt"),
```

Comment:
```suggestion
		Name:      fmt.Sprintf("%s-%s", scrt.Name, "secret"),
```

the characters saved aren't worth the added difficulty reading this :grinning:

Reply:
Doesn't exist anymore